tests/main/try/task.yaml

```diff
@@ -0,0 +1,77 @@
+summary: Check that try command works
+
+environment:
+  PORT: 8081
+  SERVICE_FILE: "./service.sh"
+  SERVICE_NAME: "test-service"
+
+prepare: |
+  echo "Given a service listening on a port"
+  echo "#!/bin/sh\nwhile true; do echo \"HTTP/1.1 200 OK\n\nok\n\" |  nc -l -p $PORT -q 1; done" > $SERVICE_FILE
+  chmod a+x $SERVICE_FILE
+  systemd-run --unit $SERVICE_NAME $SERVICE_FILE
+  while ! netstat -lnt | grep -Pq "tcp.*?:$PORT +.*?LISTEN\n*"; do sleep 0.5; done
+
+restore: |
+  systemctl stop $SERVICE_NAME
+  rm -f $SERVICE_FILE
+
+execute: |
+  echo "Given a buildable snap in a known directory"
+  SNAP_NAME="basic-binaries"
+  SNAP_PATH="../../lib/snaps/$SNAP_NAME"
+
+  echo "When try is executed on that directory"
+  snap try $SNAP_PATH
+
+  echo "Then the snap is listed as installed with try in the notes"
+  expected="(?s)Name +Version +Rev +Developer +Notes\n\
+  basic-binaries +.*?try"
+  echo "$(snap list)" | grep -Pzq "$expected"
+
+  echo "And commands from the snap-try binary can be run"
+  basic-binaries.success
+
+  echo "And commands from the snap-try binary can read in their own dir"
+  echo "$(basic-binaries.cat)" | grep -Pzq "I output myself"
+
+  echo "====================================="
+
+  echo "Given a buildable snap which access confinement-protected resources in a known directory"
+  SNAP_NAME="dev-kmsg"
+  SNAP_PATH="../../lib/snaps/$SNAP_NAME"
```

Comment:
This one is different, by default the execution of the `dev-kmsg.reader` command should be disallowed by the confinement rules, it let us exercise them. I've moved the command to the `basic-binaries.disallowed`, let me know if that's the right place.